src/CommandMap.chpl

```diff
@@ -28,13 +28,15 @@ module CommandMap {
    * This binds a server command to its corresponding function matching the standard
    * function signature & MsgTuple return type
    */
-  proc registerFunction(cmd: string, fcf: f.type) {
-    commandMap.add(cmd, fcf);
-  }
+  proc registerFunction(cmd: string, fcf: f.type, modName: string = "", line: int = -1) {
+    if commandMap.contains(cmd) {
+      writef("Warning: Command '%s' already registered. Ignoring registration from [%s:%d]", cmd, modName, line);
+    } else {
+      commandMap.add(cmd, fcf);
 
```

Comment:
Ah, this actually solves the same problem that I was fixing with https://github.com/Bears-R-Us/arkouda/pull/3424, looks good 👍